Docs visual improvement and add documentation guide for new contributor for `docs` (#4554)
cython/cython: docs/index.rst

Also see the `Cython project homepage <https://cython.org/>`_.

Cython is an optimising static compiler for both the Python programming language and
the extended Cython programming language (based on Pyrex). It makes writing C extensions
for Python as easy as Python itself.

Quickstart
-----------

:doc:`src/quickstart/index`

>    How to install Cython compiler package.

Reviewer: ```suggestion
   How to install the Cython compiler package.
```

Author: Thank you :+1: for the fix

:doc:`src/tutorial/index`

>    How to use Cython to speedup your project.

Reviewer: You are using the verb here, not the noun.
```suggestion
   How to use Cython to speed up your project.
```

Author: Thank you again, learning new thing

Guide
------

:doc:`src/userguide/index`

>    Learn to use Cython package.

Reviewer: ```suggestion
   Learn to use Cython.
```

Author: This was a placeholder that I add to give context for `src/userguide/index` link, if the suggestion good I will give it fix.

:doc:`src/reference/index`

>    How to use Cython to speedup your project.

Reviewer: We are no longer using the reference. It is only left here to keep old links alive.

Author: Thank for info, I will comment out that from `:toctree::` and add `:orphan:` to suppress warning in build time

Development
------------

:doc:`CONTRIBUTING`
   How to contribute changes to Cython.

:doc:`Changelog <src/changes>`

>    Release note of Cython development.

Reviewer: ```suggestion
   Release notes for Cython's development.
```

Author: :hammer_and_wrench: fix
